lsp-mode.el

```diff
@@ -4984,27 +5038,37 @@ Others: TRIGGER-CHARS"
                         'lsp-completion-prefix prefix)
                 (text-properties-at 0 candidate))
                ((&CompletionItem :label :insert-text? :text-edit? :insert-text-format? :additional-text-edits?)
-                item))
+                item)
+               main-edit-region)
          (cond
            (text-edit?
+            (setq main-edit-region (lsp--range-to-region (lsp:text-edit-range text-edit?)))
             (apply #'delete-region markers)
             (insert prefix)
             (lsp--apply-text-edit text-edit?))
            ((or insert-text? label)
+            (setq main-edit-region (cons start-point (point)))
             (apply #'delete-region markers)
             (insert prefix)
             (delete-region start-point (point))
             (insert (or insert-text? label))))
 
-         (when (eq insert-text-format? 2)
+         (when (equal insert-text-format? lsp/insert-text-format-snippet)
            (let (yas-indent-line)
              (yas-expand-snippet
               (lsp--to-yasnippet-snippet (buffer-substring start-point (point)))
               start-point
               (point))))
 
-         (when (and lsp-completion-enable-additional-text-edit additional-text-edits?)
-           (lsp--apply-text-edits additional-text-edits?))
+         (when lsp-completion-enable-additional-text-edit
+           (if (get-text-property 0 'lsp-completion-resolved candidate)
```

Comment:
don't we need to check if the edits are present and resolve only if they are nil?

Reply:
Fixed. I wonder if there's case of completion item need to be resolved even if additional edits are not null

Reply:
Yes. When documentation is requested. You may refer to company-lsp code - it calls resolve only if particular field is missing.